auth/src/main/java/org/aerogear/mobile/auth/authenticator/oidc/OIDCAuthenticatorImpl.java

```diff
@@ -96,11 +102,12 @@ public void authenticate(final AuthenticateOptions authOptions,
     // Authentication code
     private void performAuthRequest(final Activity fromActivity, final int resultCode) {
         nonNull(fromActivity, "fromActivity");
-
+        if (this.certificatePinningCheck.getError() != null) {
```

Comment:
The check is performed very far from this code.  This exception should probably be closer to where the check is originally made.

Reply:
As I commented above, it is possible for user not being to able to login at all due to network issues. I don't think that's right.